webhook/webhook.go

```diff
@@ -529,7 +529,15 @@ func (ac *AdmissionController) mutate(ctx context.Context, kind metav1.GroupVers
 
 	var patches []jsonpatch.JsonPatchOperation
 
-	err := updateGeneration(ctx, &patches, oldObj, newObj)
+	// Add these before defaulting fields, otherwise defaulting may cause an illegal patch because
+	// it expects the round tripped through Golang fields to be present already.
+	roundTripPatch, err := roundTripPatch(newBytes, newObj)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create patch for round tripped newBytes: %v", err)
+	}
+	patches = append(patches, roundTripPatch...)
+
+	err = updateGeneration(ctx, &patches, oldObj, newObj)
 	if err != nil {
 		logger.Error("Failed to update generation", zap.Error(err))
```

Comment:
Stupid zap logger interface :|
```suggestion
		logger.Errorw("Failed to update generation", zap.Error(err))
```

Reply:
This is why I prefer the non-sugared zap logger.